test/nodims.jl

```diff
@@ -0,0 +1,29 @@
+using Test
+using DynamicQuantities: NoDims, dimension, Quantity, UnionAbstractQuantity
+
+@testset "NoDims" begin
+    nd = NoDims()
+    @test typeof(nd) == NoDims
+end
+
+@testset "dimension" begin
+    @test dimension(1.0) == NoDims()
+    @test dimension(Quantity(1.0, length=1)) != NoDims()
```

Comment:
Should add more tests here with other quantities. Like `dimension(1.0) == dimension(u"1")`.